File: miniMaxSum/miniMAxSum.js
class miniMaxSum {

    solve( arr){
        for(let x of arr){
            if(x<1 || x>Math.pow(10,9) ){
                return "";
            }
                }
        arr = arr.sort();
        return arr.slice(0,4).reduce(this.add,0) + " " + 
        arr.slice(arr.lenght-4,arr.length).reduce(this.add,0);
        
    }

    add(accumulator,a){
        return accumulator + a;
    }

    

}


const miniMaxSum_class= new miniMaxSum();
console.log(miniMaxSum_class.solve([1,2,3,4,5]))

File: miniMaxSum/miniMaxSum.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func main() {
	prueba := []int{4, 2, 3, 1, 5}
	fmt.Println(miniMaxSum(prueba))
}

func miniMaxSum(arr []int) string {

	for _, s := range arr {
		if s < 1 || float64(s) > math.Pow(10, 9) {
			return ""
		}
	}
	sort.Ints(arr)
	min := 0
	max := 0

	for _, i := range arr[0:4] {
		min += i
	}
	for _, i := range arr[len(arr)-4 : len(arr)] {
		max += i
	}

	return fmt.Sprint(min, max, "")
}
